### PR TITLE
[RF-669] Update to new Product Onboarding Process

### PR DESCRIPTION
**Pull request by ksalmon**
## Ticket
https://gathering.atlassian.net/wiki/spaces/R3/pages/1778941972/Proposal+For+New+Onboarding+Flow+With+Multiple+Custodial+Accounts+Using+Products

Products: https://rize-developer-dev.netlify.app/#tag/Products
CustomerProducts: https://rize-developer-dev.netlify.app/#tag/Customer-Products

### New Steps!
**POST /customers with external_uid and email**

- Initializes an empty customer record

**GET /products**

- Returns the list of products available for your program and all requirments

- Not a required step if a client already knows their Product UID’s and requirements

**POST /compliance_workflows with customer_uid and compliance_plan_uid**

- Starts a compliance workflow process just like today

- Client continues acknowledging documents as necessary until complete

**PUT /customers/uid**

- Client independently updates all required PII for the Customer

**PUT /customers/uid/update_profile_responses**

- Client answers any required questions (if any) for the investor profile

- These questions are found on the individual product object

**POST /customer_products with customer_uid and product_uid**

- This effectively becomes the new identity_verification endpoint

- This will verify that all Product requirements have been met

- For the FIRST product, this will automatically kick off the KYC process

- Upon KYC approval, all required custodial accounts will be created and the customer will be active

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,20 +215,45 @@ const updatedCustomer = await rize.customer.update(
...
+        profile_response: 'yes',
+  }
+);
+```
+
+
+
+### 8. Create new Customer Product
+
+The last step will trigger verification that all Product requirements have been met. After completion Rize will automatically kick off the KYC process if it the customers first product. Upon successful create, all required custodial accounts will be created and the customer will be active. This is a billable event and is isolated intentionally for you to confirm that the Customer record is complete. ([more info](docs.md#customer))
```
Comment (RockinRonE) [inline README.md:240]: Revised the paragraph, aka, I ran it through Grammarly:

The last step will trigger verification that all Product requirements have been met. After completion, Rize will automatically kick off the KYC process if it is the customer's first product. Upon successful creation, all required custodial accounts will be created, and the customer will be active. This is a billable event and is intentionally isolated for you to confirm that the Customer record is complete.

File: README.md [hunk 2]
```diff
@@ -146,20 +215,45 @@ const updatedCustomer = await rize.customer.update(
...
-### 5. Request Identity Verification
-The Identity Verification of a customer serves as explicit confirmation from you that the
-Customer is ready for account opening. This event initiates the KYC/AML verification process
-and account opening at the Custodian(s) in your Program. This is a billable event and is isolated
-intentionally for you to confirm that the Customer record is complete. ([more info](docs.md#verifyidentity))
+
+
+### 7. Answer Profile Answers for the Plan Requirements (if needed)
+
+Each plan may come with questions that are required in order to complete the onboarding flow for the new customer. ([more info](docs.md#customer))
```
Comment (RockinRonE) [inline README.md:222]: Each plan may come with questions that are required to complete the onboarding flow for the new customer.

File: README.md [hunk 3]
```diff
@@ -1,81 +1,139 @@
...
+);
+
+console.log(newCustomer.uid); // Customer UID
+```
+
+
+
+### 2. Get Avaliable Products
+
+Next we need to retrieve a list of products available for your program and all requirements for the new customer to complete.
```
Comment (RockinRonE) [inline README.md:100]: Add a comma after Next

File: README.md [hunk 4]
```diff
@@ -1,81 +1,139 @@
...
+
 ## Quick Steps for Customer Onboarding
-### 1. Create a new Compliance Workflow.
-Compliance Workflows is the first endpoint you will use to onboard a new Customer.
-Your first request to create a compliance workflow creates a Customer on the Rize Program.
-The Customer UID returned in the response is how Rize identifies this customer on your Rize Program.
+
+### 1. Create a new Customer.
+
+Create Customer is the first endpoint you will use to start the onboard flow.
```
Comment (RockinRonE) [inline README.md:82]: Create Customer is the first endpoint to start the onboard flow. 

Active voice :)

File: README.md [hunk 5]
```diff
@@ -1,81 +1,139 @@
...
+Your first request to create empty customer with a external id and the users email and a customer object is returned with a uid.
```
Comment (RockinRonE) [inline README.md:83]: I think this needs to be rewritten. Are you saying that the first request will return a customer object that contains an external id and the user's email? 

FYI: I noticed a pattern of forgetting to have "users" as "user's" when singular.

File: README.md [hunk 6]
```diff
@@ -1,81 +1,139 @@
...
+const productList = await rize.product.getList();
+
+console.log(productList[0].uid); // First product UID
+```
+
+
+
+### 3. Create a new Compliance Workflow.
+
+Creating a compliance workflow just requires the new customer UID and the compliance plan UID from the last two steps.
```
Comment (RockinRonE) [inline README.md:115]: I would remove "just" as it's not necessary

File: README.md [hunk 7]
```diff
@@ -1,81 +1,139 @@
...
     'client-generated-external-uid-42',
-    'tomas@example.com'
+    'compliance-plan-92'
 );
 
 console.log(complianceWorkflow.uid); // Workflow UID
-console.log(complianceWorkflow.customer.uid); // Customer UID
 ```
+
 Your request to the Compliance workflows endpoint initiates a Compliance Workflow for the
```
Comment (RockinRonE) [inline README.md:128]: Sending a request to the Compliance workflow's endpoint initiates a Compliance Workflow for the onboarding customer. The Compliance Workflow contains disclosure and compliance documents that the Customer must acknowledge before opening a Custodial
Account.

File: README.md [hunk 8]
```diff
@@ -169,21 +263,29 @@ In the Rize Sandbox, you can generate and test your receipt of the customer eval
 statuses using the last names supplied with the Customer record. The statuses in the Rize
```
Comment (RockinRonE) [inline README.md:263]: What is "undergo a statuses" ?

File: README.md [hunk 9]
```diff
@@ -169,21 +263,29 @@ In the Rize Sandbox, you can generate and test your receipt of the customer eval
...
-| Smith                                  | Manual Review                  | The Custodian is unable to onboard the Customer through an automated process and additional information is required.
+| Customer Last Name                     | Customer KYC_Status Returned | Description                                                  |
+| -------------------------------------- | ---------------------------- | ------------------------------------------------------------ |
+| Any value, excluding Smith and Johnson | Approved                     | Approved status indicates that the Custodian will allow the Customer to access their Service Offering immediately. The Customer has completed onboarding to the Program. No documentation is required. |
+| Johnson                                | Denied                       | The Customer is denied access to the Custodian’s products. No recourse is available through the Custodian or Rize. |
+| Smith                                  | Manual Review                | The Custodian is unable to onboard the Customer through an automated process and additional information is required. |
```
Comment (RockinRonE) [inline README.md:270]: Comma before "and"

**Review by RockinRonE — CHANGES_REQUESTED** [review]
Some grammar updates.

Appreciate all the documentation!

File: README.md [hunk 10]
```diff
@@ -1,81 +1,139 @@
...
 ```
-| Option        | Description                                                                     | Default  |
-| ------------- | --------------------------------------------------------------------------------| -------- |
-| environment   | The Rize environment to be used: `'sandbox'`, `'integration'` or `'production'` | 'sandbox'|
-| timeout       | The number of milliseconds before the each request times out                    | 80000    |
+
+| Option      | Description                                                  | Default   |
+| ----------- | ------------------------------------------------------------ | --------- |
+| environment | The Rize environment to be used: `'sandbox'`, `'integration'` or `'production'` | 'sandbox' |
+| timeout     | The number of milliseconds before the each request times out | 80000     |
```
Comment (KramerJeff) [inline README.md:68]: Should be: "The number of milliseconds before each request times out"

File: lib/core/customer-product.js [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+        const queryStr = utils.toQueryString(query, queryParameters);
+
+        const response = await this._api.get(`/customer_products/${queryStr}`);
+
+        return response.data;
+    }
+
+    /**
+     * Creates a new instance of a customer product. This will verify that all Product requirements have been met.
+     * After completion Rize will automatically kick off the KYC process if it the customers first product.
```
Comment (KramerJeff) [inline lib/core/customer-product.js:113]: After completion, Rize will automatically kick off the KYC process if it is the customer's first product.

**Review by KramerJeff — CHANGES_REQUESTED** [review]

**Review by KramerJeff — APPROVED** [review]

**Comment by KramerJeff** [comment]
Grammar changes are all done, good work @ksalmon

**Pull request merged**